### ISSUE TITLE
QuintIR  -> ApalacheIR

### ISSUE THREAD

**Issue by shonfeder** [informalsystems/apalache#2350]
We need the type of operator declarations before we can convert them into tla
operator declarations.

[`TlaOperDecl`](https://github.com/informalsystems/apalache/blob/79e2e0c88ad6c121e1f8b027d74d163793918f29/tlair/src/main/scala/at/forsyte/apalache/tla/lir/TlaDecl.scala#L87-L88) requires a list of [OperParam](https://github.com/informalsystems/apalache/blob/79e2e0c88ad6c121e1f8b027d74d163793918f29/tlair/src/main/scala/at/forsyte/apalache/tla/lir/OperParam.scala#L8)s.
These are just pairs of the paramter name and it's arity (values have arity
0). To determine the arity of a parameter to a tnt operator def will require
inspecting the type inferred/checked for the operator. 

The [scoped builder also requires each parameter to be typed
separately](https://github.com/informalsystems/apalache/blob/79e2e0c88ad6c121e1f8b027d74d163793918f29/tlair/src/main/scala/at/forsyte/apalache/tla/typecomp/ScopedBuilder.scala#L236-L237), which again means 
we'll need to inspect an operator declaration's type recover
the types for each parameter.

We cannot construct TLA operator declarations until we have the
types of the Quint operator declarations available. Whereas type annotations for
variables and constants in required, they are optional for operator
declarations, so as a rule we are not getting the types in the AST.

Thus, the following are pre-requisites for this issue

- [x] informalsystems/quint#345 
- [x] informalsystems/quint#349
- [x] #2351
- [ ] #2363
- [ ] #2364
- [ ] #2365

**Comment by shonfeder** [comment]
this has become superseded by the factored Out issues, and we don't need it for tracking when we have the milestone
